setup.py

```diff
@@ -542,7 +542,7 @@ def check_file(f):
     cuda_include_path = os.path.join(CUDA_HOME, 'include')
     include_dirs.append(cuda_include_path)
     include_dirs.append(tmp_install_path + "/include/THCUNN")
-    extra_compile_args += ['-DWITH_CUDA']
+    extra_compile_args += ['-DWITH_CUDA', '-DAT_CUDA_ENABLED']
```

Comment:
btw how is this related to this PR?

Reply:
Somehow AT_CUDA_ENABLED got added into ATen headers so it has to be defined if you are including them. I have to go stamp that out, the headers shouldn't depend on cuda.

Reply:
What do you mean by "it got added to ATen headers"? We're adding it unconditionally and this is quite worrying

Reply:
I mean that someone made the headers depend on AT_CUDA_ENABLED being set or not. This is not good, because consumers of AT need to set it in the same way that ATen was built with. We need to remove AT_CUDA_ENABLED from all ATen public headers.

For now, I set it with WITH_CUDA so that it is at least correct for now.

Reply:
See also https://github.com/zdevito/ATen/issues/168; I have a fix for this in my giant CuDNN refactor PR (and it might even be a single, extricable commit!)